modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/field/predicate/provider/KeywordsFieldPredicateProvider.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.rest.internal.odata.filter.expression.field.predicate.provider;
```

Comment:
I'm seeing that the implementations are also in `object-rest-impl` instead of `object-service`.

For the relationships, as it is supposed to return a Predicate, all of the implementations are on the service module, I don't if we should follow the same pattern or just leave it as it is.

Reply:
The problem is the same as above. The interface is coupled to odata, so we would be coupling the service to odata too...